File: Java/Program/Komputer.java
// Komputer.java
import java.util.List;
import java.util.ArrayList;

public class Komputer {
    private String nama;
    private Cpu cpu;
    private Gpu gpu;
    private List<Ram> ramList;
    private List<Storage> storageList;
    
    public Komputer() {
        this.ramList = new ArrayList<>();
        this.storageList = new ArrayList<>();
    }
    
    public Komputer(String nama, Cpu cpu, Gpu gpu, List<Ram> ramList, List<Storage> storageList) {
        this.nama = nama;
        this.cpu = cpu;
        this.gpu = gpu;
        this.ramList = ramList;
        this.storageList = storageList;
    }
    
    // Setter methods
    public void setNama(String nama) {this.nama = nama;}
    public void setCpu(Cpu cpu) {this.cpu = cpu;}
    public void setGpu(Gpu gpu) {this.gpu = gpu;}
    public void setRamList(List<Ram> ramList) {this.ramList = ramList;}
    public void setStorageList(List<Storage> storageList) {this.storageList = storageList;}
    
    // Getter methods
    public String getNama() {return nama;}
    public Cpu getCpu() {return cpu;}
    public Gpu getGpu() {return gpu;}
    public List<Ram> getRamList() {return ramList;}
    
    public List<Storage> getStorageList() {
        return storageList;
    }
    
    public void displayInfo() {
        System.out.println("--------------------------------------------------");
        System.out.println("             Informasi Komputer");
        System.out.println("--------------------------------------------------");
        System.out.println("Nama: " + nama);
        
        System.out.print("CPU: ");
        cpu.displayInfo();
        
        System.out.print("GPU: ");
        gpu.displayInfo();
        
        System.out.println("RAMs: ");
        for (Ram ram : ramList) {
            System.out.print("\t");
            ram.displayInfo();
        }
        
        System.out.println("Storages: ");
        for (Storage storage : storageList) {
            System.out.print("\t");
            storage.displayInfo();  // Polymorphic
        }
    }
}

File: Python/program/Komputer.py
from Cpu import Cpu
from Gpu import Gpu
from Ram import Ram
from Storage import Storage
from Hdd import Hdd
from Ssd import Ssd

class Komputer:
    def __init__(self, nama, cpu, gpu, ramList, storageList):
        self.nama = nama
        self.cpu = cpu
        self.gpu = gpu
        self.ramList = ramList
        self.storageList = storageList

    # Setter
    def set_nama(self, nama):
        self.nama = nama

    def set_cpu(self, cpu):
        self.cpu = cpu

    def set_gpu(self, gpu):
        self.gpu = gpu

    def set_ram_list(self, ramList):
        self.ramList = ramList

    def set_storage_list(self, storageList):
        self.storageList = storageList

    # Getter
    def get_nama(self):
        return self.nama

    def get_cpu(self):
        return self.cpu

    def get_gpu(self):
        return self.gpu

    def get_ram_list(self):
        return self.ramList

    def get_storage_list(self):
        return self.storageList

    def display_info(self):
        print("--------------------------------------------------")
        print("             Informasi Komputer")
        print("--------------------------------------------------")
        print(f"Nama: {self.nama}")

        print("CPU: ", end="")
        self.cpu.displayInfo()

        print("GPU: ", end="")
        self.gpu.displayInfo()

        print("RAMs: ")
        for ram in self.ramList:
            print("\t", end="")
            ram.displayInfo()

        print("Storages: ")
        for storage in self.storageList:
            print("\t", end="")
            storage.displayInfo()
